app/src/ui/diff/side-by-side-diff.tsx

```diff
@@ -657,16 +657,72 @@ export class SideBySideDiff extends React.Component<
 
     const { from, to } = range
 
+    const { lineNumbers, lineNumbersIdentifiers, diffType } =
+      this.getRowGroupLineNumberData(row.hunkStartLine)
     return {
       isFirst: prev === undefined || !selectableType.includes(prev.type),
       isLast: next === undefined || !selectableType.includes(next.type),
       isHovered: hoveredHunk === row.hunkStartLine,
       isFocused: true, // focusedHunk === row.hunkStartLine, - To be added in later PR
       selectionState: selection.isRangeSelected(from, to - from + 1),
       height: this.getRowSelectableGroupHeight(row.hunkStartLine),
+      lineNumbers: Array.from(lineNumbers),
+      lineNumbersIdentifiers,
+      diffType,
     }
   }
 
+  private getRowGroupLineNumberData = (hunkStartLine: number) => {
+    const rows = getDiffRows(
+      this.state.diff,
+      this.props.showSideBySideDiff,
+      this.canExpandDiff()
+    )
+
+    const lineNumbers = new Set<number>()
+    let hasAfter = false
+    let hasBefore = false
+
+    const lineNumbersIdentifiers = rows.flatMap(r => {
```

Comment:
TLDR: please review for logic flaws, but also know that I am looking to iterate on this portion to some extent.

Going to open another PR for figuring out how to display the hunk handle even when the first hunk row is not in the virtual list rendering. My current thinking is to try to calculate `isFirst` with different metric... something like the "prev.type isn't selectable" || "is first renderable row in group" based on how many rows the virtual list actually renders... This would also impact the height of the row by however many rows in the group are not renderable.

In that PR, I am also going to look at making this whole thing more performant as a regression to this code is that files with a lot of file changes take longer to load when scrolling quickly. One thing I know off the bat is the loop I am doing here is very similar to the loop I am doing is very similar to the loop I do in `getRowSelectableGroupHeight` so maybe they can be combined. Additionally, both of these will loop through all rows, should be able to make it cut the loop short once the row's hunk data is found. Secondly, the line number  and line number identifiers and modified stuff I don't think should change after initial render.. so maybe we can cache some of this info instead of looping through all the rows on each row render.. not sure about row height tho because it would need to be cleared on a resize event.  Another possible performance enhancement is if the row group height is greater than the number of virtual rows rendered, then stop looping because it we never need to render stuff for rows that can't be visually seen. (like don't render a 5000px hunk handle on a screen that can only show 1000px).